src/Pages/OnePostContent/ContentPage.tsx

```diff
@@ -1,58 +1,108 @@
 import { useEffect } from "react";
-import { GeneralPage } from "../../components/GeneralPage/GeneralPage";
 import { Button } from "../../components/Button";
 import { Prev } from "../../components/NavForPage/Prev/Prev";
 import { Next } from "../../components/NavForPage/Next/Next";
 import "./ContentPage.scss";
 import { Link, useParams } from "react-router-dom";
 import { useDispatch, useSelector } from "react-redux";
-
 import { getOnePost } from "../../Redux/singlePost/singePost";
 import { Loader } from "../../components/Loader/Loader";
-
+import { GeneralPage } from "Pages/GeneralPage/GeneralPage";
+import { ReactComponent as Like } from "../../components/image/like.svg";
+import { ReactComponent as SvgBwi } from "../../components/image/buttonWithIcon.svg";
+import { ReactComponent as Dislike } from "../../components/image/dislike.svg";
+import {
+  dislikePost,
+  favoritPost,
+  likePost,
+} from "Redux/posts/PostsActionReducer";
 export const ContentPage = () => {
   const { id } = useParams();
-  const post = useSelector((state: any) => state.singlepostReducer.post);
-  console.log(post);
-
   const dispatch = useDispatch();
+  const postsLike = useSelector(
+    (state: any) => state.postReducer.content
+  )?.find((el: any) => el.id === Number(id));
```

Comment:
Раздели получение всех постов и получение поста с лайком
```suggestion
  const posts = useSelector((state: any) => state.postReducer.content)
  const postLike = posts?.find((el: any) => el.id === Number(id))
```